signer/src/wsts_state_machine.rs

```diff
@@ -30,33 +39,435 @@ use wsts::traits::Signer as _;
 /// hash bytes while for the signing rounds we identify the state machine
 /// by the sighash bytes.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct StateMachineId([u8; 32]);
+pub enum StateMachineId {
+    /// Identifier for a DKG state machines
+    Dkg(model::BitcoinBlockHash),
+    /// Identifier for a Bitcoin signing state machines
+    BitcoinSign(SigHash),
+    /// Identifier for a rotate key verification signing round
+    RotateKey(PublicKeyXOnly, model::BitcoinBlockHash),
+}
 
 impl From<&model::BitcoinBlockHash> for StateMachineId {
     fn from(value: &model::BitcoinBlockHash) -> Self {
-        StateMachineId(value.to_byte_array())
+        StateMachineId::Dkg(*value)
     }
 }
 
 impl From<SigHash> for StateMachineId {
     fn from(value: SigHash) -> Self {
-        StateMachineId(value.to_byte_array())
+        StateMachineId::BitcoinSign(value)
+    }
+}
+
+/// A trait for converting a message into another type.
+pub trait FromMessage {
+    /// Convert the given message into the implementing type.
+    fn from_message(message: &Message) -> Self
+    where
+        Self: Sized;
+}
+
+impl FromMessage for Packet {
+    fn from_message(message: &Message) -> Self {
+        Packet {
+            msg: message.clone(),
+            sig: Vec::new(),
```

Comment:
this is a bit strange, does it mean we never care about those signatures when processing the packages?

Reply:
Yeah, I reacted on it too. This is just a helper method to avoid doing the `Packet { .. }` elsewhere, where originally calling code would create the `wsts::Packet` from a `wsts::Mesage` in the above style and call `process_packet()`. So the actual logic shouldn't be changed; it should just be a little cleaner from the caller's perspective.

That's also where the new `process_message()` on the `WstsCoordinator` trait came from as well -- the wsts `process_message()` takes a `Packet`. So the new method instead takes a `Message` and does the redirection to the oddly named wsts `process_message(Packet)`.

@xoloki could check and confirm that there's nothing weird here